org.bndtools.applaunch/src/org/bndtools/applaunch/Activator.java

```diff
@@ -69,6 +75,39 @@ private static Filter createFilter() {
 		return context.registerService(ApplicationLauncher.class, new BndApplicationLauncher(context), null);
 	}
 
+	private void setEquinoxConfigAppArgs(BundleContext context) {
+		String[] args = null;
+		try {
+			ServiceReference<?>[] serviceReferences = context.getServiceReferences("aQute.launcher.Launcher",
+				"(launcher.arguments=*)");
+			if (serviceReferences != null && serviceReferences.length > 0) {
+				Object property = serviceReferences[0].getProperty("launcher.arguments");
+
+				if (property instanceof String[]) {
+					args = (String[]) property;
+
+					ServiceReference<EnvironmentInfo> srvcRef = context.getServiceReference(EnvironmentInfo.class);
+					if (srvcRef != null) {
+						EnvironmentInfo service = context.getService(srvcRef);
+						if (service instanceof EquinoxConfiguration) {
+							EquinoxConfiguration equinoxConfig = (EquinoxConfiguration) service;
+							equinoxConfig.setAllArgs(args);
+							equinoxConfig.setAppArgs(args);
+						}
+					}
+
+					log.fine("configured program arguments " + Arrays.stream(args)
```

Comment:
I think `Arrays.toString(args)` is probably fine and simpler.

Why is this log `log.fine(` when the other logs are like `log.log(Level.`? It seems a bit inconsistent.

Reply:
I'll clean those up.